# Creative Capitol FEC Filings

Using FEC Disbursements to look at the cost of creative services in U.S. Political Campaigns

##### Imports

In [1]:
#Import libraries
import pandas as pd

/Users/sm/.pyenv/versions/3.10.0/lib/python3.10/site-packages/pandas/compat/__init__.py:124: UserWarning: Could not import the lzma module. Your installed Python is incomplete. Attempting to use lzma compression will result in a RuntimeError.
  warnings.warn(msg)


In [2]:
import numpy as np
from matplotlib import pyplot as plt

In [3]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
pd.set_option('display.width', None)
pd.set_option('display.max_colwidth', None)
pd.set_option('max_seq_item', None)

##### FEC API

https://api.open.fec.gov/developers/

https://api.open.fec.gov/developers/#/search

API Key = 1d4FL1ITomWsY51hgi0HjixVu6J0yMzWB4ThdLgI

Disbursement Categories 
https://www.fec.gov/campaign-finance-data/disbursement-category-code-descriptions/

Disbursement category codes for non-presidential filers:

* 001	Administrative/salary/overhead expenses (e.g., rent, staff salaries, postage, office supplies, equipment, furniture, ballot access fees, petition drives, party fees and legal and accounting expenses)
* 003	Solicitation and fundraising expenses (e.g., costs for direct mail solicitations and fundraising events including printing, mailing lists, consultant fees, call lists, invitations, catering costs and room rental)
* 004	Advertising expenses -including general public political advertising (e.g., purchases of radio/television broadcast/cable time, print advertisements and related production costs)
* 006	Campaign materials (e.g., buttons, bumper stickers, brochures, mass mailings, pens, posters, balloons)


Disbursement category codes for presidential filers:
* 102	Media expenditures
* 103	Expenditures for mass mailings and other campaign materials (e.g., buttons, bumper stickers, brochures, mass mailings, pens, posters, balloons)
* 104	Overhead expenditures of state offices and their facilities (e.g., rent, staff salaries, postage, office supplies, equipment, furniture, ballot access fees, petition drives, party fees and legal and accounting expenses)
* 107	Fundraising expenditures (e.g., costs for direct mail solicitations & fundraising events including printing, mailing lists, consultant fees, call lists, invitations, catering costs and room rental)




FEC Disbursement Description Search Example: https://www.fec.gov/data/disbursements/?spender_committee_type=S&data_type=processed&max_date=03%2F06%2F2022&disbursement_description=graphics&disbursement_description=graphic

##### Examples of Creative Related Expenses by Campaigns

Goal: Determine potential query terms for the FEC database by researching terms used by a known creative agency, Wide Eye, a disbursement recipient doing campaign graphics work 

In [4]:
df_wideeye = pd.read_csv("..//data//source//FEC//schedule_b-2022-03-06T09_25_48_wideeye.csv", encoding="utf-8")

In [5]:
df_wideeye.head(20)

committee_id               committee_name  report_year report_type  \
0     C00193433                 EMILY'S LIST         2021          YE   
1     C00495861        PRIORITIES USA ACTION         2021          YE   
2     C00574889           TAMMY FOR ILLINOIS         2021          Q1   
3     C00344473       STABENOW FOR US SENATE         2021          YE   
4     C00588772                MAGGIE FOR NH         2021          Q1   
5     C00588772                MAGGIE FOR NH         2021          Q1   
6     C00639054     DEB HAALAND FOR CONGRESS         2021          Q1   
7     C00341396  MOVEON.ORG POLITICAL ACTION         2021         30R   
8     C00696526        MARK KELLY FOR SENATE         2021          Q1   
9     C00495358           KAINE FOR VIRGINIA         2021          Q1   
10    C00709899        SARA GIDEON FOR MAINE         2021          Q3   
11    C00770602           RHODA FOR CONGRESS         2021          Q2   
12    C00341396  MOVEON.ORG POLITICAL ACTION         2021         12S   
13    C00341396  MOVEON.ORG POLITICAL ACTION         2021         12S   
14    C00431304               GET STUFF DONE         2021          MY   
15    C00327023      SCHAKOWSKY FOR CONGRESS         2021          Q1   
16    C00574889           TAMMY FOR ILLINOIS         2021          Q2   
17    C00341396  MOVEON.ORG POLITICAL ACTION         2021          MY   
18    C00341396  MOVEON.ORG POLITICAL ACTION         2021          MY   
19    C00341396  MOVEON.ORG POLITICAL ACTION         2021          MY   

          image_number line_number transaction_id  file_number entity_type  \
0   202201259475150998         21B   SB21B-376250      1558915         ORG   
1   202202029491531310          29    VNTYHA2J5N0      1567128         ORG   
2   202104159443429020          17    VN7ME9ZV2M2      1511405         ORG   
3   202201319485206062          17      500055632      1563121         ORG   
4   202104159443584902          17    VPED5A9E546      1512026         ORG   
5   202104159443584902          17    VPED5A9E554      1512026         ORG   
6   202104159443425710          17    VNGZH9ZTR96      1511395         ORG   
7   202104079443135272         21B      500016476      1508413         ORG   
8   202104159443656829          17    VVB3SAQSZR8      1512346         ORG   
9   202104159443367762          17        D677719      1510947         ORG   
10  202112089469819098          17    VVB9ZAQYV81      1551022         ORG   
11  202112089469819801          17       5498919I      1551057         ORG   
12  202112029469781465         21B      500022909      1550565         ORG   
13  202112029469781466         21B      500022908      1550565         ORG   
14  202107199451681624         21B    VTDBJAET1C4      1528430         ORG   
15  202107209451842051          17      SB17.4788      1529283         ORG   
16  202107159451323414          17    VN7ME9ZX007      1526447         ORG   
17  202107319465462951         21B      500017653      1533181         ORG   
18  202107319465462951         21B      500017651      1533181         ORG   
19  202107319465462951         21B      500017652      1533181         ORG   

   entity_type_desc  unused_recipient_committee_id  recipient_committee_id  \
0      ORGANIZATION                            NaN                     NaN   
1      ORGANIZATION                            NaN                     NaN   
2      ORGANIZATION                            NaN                     NaN   
3      ORGANIZATION                            NaN                     NaN   
4      ORGANIZATION                            NaN                     NaN   
5      ORGANIZATION                            NaN                     NaN   
6      ORGANIZATION                            NaN                     NaN   
7      ORGANIZATION                            NaN                     NaN   
8      ORGANIZATION                            NaN                     NaN   
9      ORGANIZATION                            NaN

In [6]:
df_wideeye.disbursement_description

0                                   WEBSITE DEVELOPMENT & DESIGN
1                  NON-CONTRIBUTION ACCOUNT: WEBSITE DEVELOPMENT
2                                               WEBSITE SERVICES
3                                                        WEBSITE
4                                                WEBSITE HOSTING
5                                                WEBSITE HOSTING
6                                                     WEB DESIGN
7                                 PROGRAM DEVELOPMENT CONSULTANT
8                                               WEBSITE SERVICES
9                                                WEBSITE SUPPORT
10                                           WEBSITE DEVELOPMENT
11    FUNDRAISING VENUE INCLUDING BARTENDER AND BARISTA SERVICES
12                                PROGRAM DEVELOPMENT CONSULTANT
13                                PROGRAM DEVELOPMENT CONSULTANT
14                                                   WEB HOSTING
15                       

In [7]:
df_wideeye.disbursement_description.unique()

array(['WEBSITE DEVELOPMENT & DESIGN',
       'NON-CONTRIBUTION ACCOUNT: WEBSITE DEVELOPMENT',
       'WEBSITE SERVICES', 'WEBSITE', 'WEBSITE HOSTING', 'WEB DESIGN',
       'PROGRAM DEVELOPMENT CONSULTANT', 'WEBSITE SUPPORT',
       'WEBSITE DEVELOPMENT',
       'FUNDRAISING VENUE INCLUDING BARTENDER AND BARISTA SERVICES',
       'WEB HOSTING', 'WEBSITE MAINTENANCE',
       'WEBSITE HOSTING-REPLACES CHECK #25206'], dtype=object)

'WEBSITE DEVELOPMENT & DESIGN',
       'NON-CONTRIBUTION ACCOUNT: WEBSITE DEVELOPMENT',
       'WEBSITE SERVICES', 
       'WEBSITE', 
       'WEBSITE HOSTING', 
       'WEB DESIGN',
       'PROGRAM DEVELOPMENT CONSULTANT', 
       'WEBSITE SUPPORT',
       'WEBSITE DEVELOPMENT',
       'FUNDRAISING VENUE INCLUDING BARTENDER AND BARISTA SERVICES',
       'WEB HOSTING', 
       'WEBSITE MAINTENANCE',
       'WEBSITE HOSTING-REPLACES CHECK #25206']

Disbursements for Website-related Services

In [8]:
df_website = pd.read_csv("..//data//source//FEC//schedule_b-2022-03-06T11_19_30_website.csv")

/Users/sm/.pyenv/versions/3.10.0/lib/python3.10/site-packages/IPython/core/interactiveshell.py:3457: DtypeWarning: Columns (15,31,32,50,54,64,74,75,77) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [9]:
df_website.head()

committee_id                              committee_name  report_year  \
0    C00394395                       HUNZEKER FOR CONGRESS         2004   
1    C00300921  BLACK AMERICA'S POLITICAL ACTION COMMITTEE         2004   
2    C00399907                       NEWBERRY FOR CONGRESS         2004   
3    C00365015            COMMITTEE TO ELECT LINCOLN DAVIS         2004   
4    C00392266                      STREUSAND FOR CONGRESS         2004   

  report_type  image_number line_number transaction_id  file_number  \
0          Q2   24962140808          17       EXP:B:71       131870   
1          Q3   25980146785         21B    SB21B.37070       158644   
2         12P   24961973931          17      SB17.4856       129227   
3          Q3   24981789481          17     SB17.13726       156587   
4         12R   24990911003          17      SB17.4601       113365   

  entity_type entity_type_desc unused_recipient_committee_id  \
0         ORG     ORGANIZATION                           NaN   
1         ORG     ORGANIZATION                           NaN   
2         IND       INDIVIDUAL                           NaN   
3         IND       INDIVIDUAL                           NaN   
4         ORG     ORGANIZATION                           NaN   

  recipient_committee_id                    recipient_name recipient_state  \
0                    NaN  DESIGNED INTERNET SOLUTIONS INC.              CA   
1                    NaN                          NGEN LLC              DC   
2                    NaN                NEWBERRY, STIRLING              MA   
3                    NaN                       HAYDEN, TOM              TN   
4                    NaN                     IP PROMOTIONS              TX   

  beneficiary_committee_name national_committee_nonfederal_account  \
0                        NaN                                   NaN   
1                        NaN                                   NaN   
2                        NaN                                   NaN   
3                        NaN                                   NaN   
4                        NaN                                   NaN   

  disbursement_type disbursement_type_description  \
0               NaN                           NaN   
1               NaN                           NaN   
2               NaN                           NaN   
3               NaN                           NaN   
4               NaN                           NaN   

              disbursement_description memo_code  memo_code_full  \
0                              WEBSITE       NaN             NaN   
1         WEBSITE DEVELOPMENT SERVICES       NaN             NaN   
2  WEBSITE DEVELOPMENT AND MAINTENANCE       NaN             NaN   
3                           WEB DESIGN       NaN             NaN   
4                           WEB DESIGN       NaN             NaN   

     disbursement_date  disbursement_amount candidate_office  \
0  2004-05-25 00:00:00               100.00              NaN   
1  2004-09-24 00:00:00                38.33              NaN   
2  2004-07-06 00:00:00              1300.00              NaN   
3  2004-09-01 00:00:00              1000.00                H   
4  2004-03-19 00:00:00              1597.60                H   

  candidate_office_description  candidate_office_district candidate_id  \
0                          NaN                        NaN          NaN   
1                          NaN                        NaN          NaN   
2                          NaN                        NaN          NaN   
3                        HOUSE                        4.0    H4TN06051   
4                        HOUSE                       10.0          NaN   

           candidate_name candidate_first_name candidate_last_name  \
0                     NaN                  NaN                 NaN   
1                     NaN                  NaN                 NaN   
2                     NaN                  NaN                 NaN   
3         D

In [25]:
len(df_website)

93589

Disbursements for Logo-related Services

In [10]:
df_logo = pd.read_csv("..//data//source//FEC//schedule_b-2022-03-06T11_45_52_logo.csv")

In [11]:
df_logo.head()

committee_id                                       committee_name  \
0    C00153379  SIERRA PACIFIC EMPLOYEES POLITICAL ACTION COMMITTEE   
1    C00394833                             SANDY LYONS FOR CONGRESS   
2    C00254185                    GENE GREEN CONGRESSIONAL CAMPAIGN   
3    C00401703                                  FRIENDS OF ALJANICH   
4    C00385955                                 HERLONG FOR CONGRESS   

   report_year report_type  image_number line_number transaction_id  \
0         2004          Q1   24990974223         21B     SB21B.4326   
1         2004          Q1   24990967119          17      SB17.4288   
2         2004          Q2   24961810449          17      SB17.9170   
3         2004          Q2   24971485699          17      SB17.4307   
4         2003         TER   23991255864          21      SB21.4500   

   file_number entity_type entity_type_desc unused_recipient_committee_id  \
0       115064         ORG     ORGANIZATION                           NaN   
1       114815         ORG     ORGANIZATION                           NaN   
2       126632         ORG     ORGANIZATION                           NaN   
3       135364         ORG     ORGANIZATION                           NaN   
4        88691         IND       INDIVIDUAL                           NaN   

  recipient_committee_id              recipient_name recipient_state  \
0                    NaN          LYNETTE A/ STRYKER              NV   
1                    NaN  LISELLA PUBLIC AFFAIRS LLC              SC   
2                    NaN            J. HARDING & CO.              TX   
3                    NaN                   FLY PAPER              LA   
4                    NaN       WILLIAMS, ELIZABETH                SC   

  beneficiary_committee_name  national_committee_nonfederal_account  \
0                        NaN                                    NaN   
1                        NaN                                    NaN   
2                        NaN                                    NaN   
3                        NaN                                    NaN   
4                        NaN                                    NaN   

  disbursement_type disbursement_type_description  \
0               NaN                           NaN   
1               NaN                           NaN   
2               NaN                           NaN   
3               NaN                           NaN   
4               NaN                           NaN   

          disbursement_description memo_code  memo_code_full  \
0  LOGO CREATION/BROCHURE ART WORK       NaN             NaN   
1                    LETTER & LOGO       NaN             NaN   
2                 STAFF LOGO SHIRT       NaN             NaN   
3                  LOGO, BROCHURES       NaN             NaN   
4                      DESIGN LOGO       NaN             NaN   

     disbursement_date  disbursement_amount candidate_office  \
0  2004-03-09 00:00:00              1850.00              NaN   
1  2004-02-07 00:00:00              8146.31              NaN   
2  2004-06-10 00:00:00                23.71              NaN   
3  2004-06-14 00:00:00              1350.00                H   
4  2003-05-08 00:00:00              1000.00              NaN   

  candidate_office_description  candidate_office_district candidate_id  \
0                          NaN                        NaN          NaN   
1                          NaN                        NaN          NaN   
2                          NaN                        NaN          NaN   
3                        HOUSE                        3.0          NaN   
4                          NaN                        NaN          NaN   

               candidate_name candidate_first_name candidate_last_name  \
0                         NaN                  NaN                 NaN   
1                         NaN                  NaN                 NaN   
2                         NaN                  NaN                 Na

In [24]:
# df_logo.disbursement_description

In [23]:
len(df_logo)

2742

Disbursement for Design-related Services

In [13]:
df_design = pd.read_csv("..//data//source//FEC//schedule_b-2022-03-06T11_50_56_design.csv")

/Users/sm/.pyenv/versions/3.10.0/lib/python3.10/site-packages/IPython/core/interactiveshell.py:3457: DtypeWarning: Columns (15,64,74,75,76,77) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [14]:
df_design.head()

committee_id            committee_name  report_year report_type  \
0    C00564948   PAUL SPAIN FOR CONGRESS         2015          YE   
1    C00583005  JESSE SBAIH FOR CONGRESS         2015          YE   
2    C00505776       FRIENDS OF MIA LOVE         2016         12G   
3    C00566174   TEA PARTY MAJORITY FUND         2016          Q1   
4    C00620369     HOUSE LIBERTY PROJECT         2016         30G   

         image_number line_number transaction_id  file_number entity_type  \
0  201601149004484378          17      SB17.4691      1039612         ORG   
1  201601139004479498          17    VQZA8A1G006      1039411         IND   
2  201701179041407967          17     SB17.I3356      1139302         ORG   
3  201910029163583164         21B     SB21B.4235      1354442         ORG   
4  201612089039991452         21B     SB21B.4239      1133212         ORG   

  entity_type_desc unused_recipient_committee_id recipient_committee_id  \
0     ORGANIZATION                           NaN                    NaN   
1       INDIVIDUAL                           NaN                    NaN   
2     ORGANIZATION                           NaN                    NaN   
3     ORGANIZATION                           NaN                    NaN   
4     ORGANIZATION                     C00620369                    NaN   

             recipient_name recipient_state beneficiary_committee_name  \
0  POLITICAL CONSULTING LLC              FL                        NaN   
1             RENFRO, ANGIE              IL                        NaN   
2      VISUAL IMPACT DESIGN              GA                        NaN   
3     IMAGINE IT DESIGN LLC              LA                        NaN   
4            MEDIA 3 DESIGN              MI      HOUSE LIBERTY PROJECT   

  national_committee_nonfederal_account disbursement_type  \
0                                   NaN               NaN   
1                                   NaN               NaN   
2                                   NaN               NaN   
3                                   NaN               NaN   
4                                   NaN               NaN   

  disbursement_type_description          disbursement_description memo_code  \
0                           NaN                        WEB DESIGN       NaN   
1                           NaN                    GRAPHIC DESIGN       NaN   
2                           NaN                    GRAPHIC DESIGN       NaN   
3                           NaN  WEBSITE UPDATES & GRAPHIC DESIGN       NaN   
4                           NaN                    GRAPHIC DESIGN       NaN   

   memo_code_full    disbursement_date  disbursement_amount candidate_office  \
0             NaN  2015-11-20 00:00:00                300.0              NaN   
1             NaN  2015-10-30 00:00:00                350.0              NaN   
2             NaN  2016-10-06 00:00:00               4000.0              NaN   
3             NaN  2016-03-10 00:00:00               1640.0              NaN   
4             NaN  2016-11-28 00:00:00                545.7              NaN   

  candidate_office_description  candidate_office_district candidate_id  \
0                          NaN                        NaN          NaN   
1                          NaN                        NaN          NaN   
2                          NaN                        NaN          NaN   
3                          NaN                        NaN          NaN   
4                          NaN                        NaN          NaN   

  candidate_name candidate_first_name candidate_last_name  \
0            NaN                  NaN                 NaN   
1            NaN                  NaN                 NaN   
2            NaN                  NaN                 NaN   
3            NaN                  NaN                 NaN   
4            NaN                  NaN                 NaN   

  candidate_middle_name candidate_prefix candidate_suffix  \
0                   NaN         

In [22]:
len(df_design)

47244

Disbursement for Graphic-related or Graphics-related Services

In [18]:
df_graphic = pd.read_csv("..//data//source//FEC//schedule_b-2022-03-07T12_54_12_graphic.csv")

/Users/sm/.pyenv/versions/3.10.0/lib/python3.10/site-packages/IPython/core/interactiveshell.py:3457: DtypeWarning: Columns (32,64) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [20]:
df_graphic.head()

committee_id                               committee_name  report_year  \
0    C00343475                       TODD AKIN FOR CONGRESS         2005   
1    C00396325  COMMITTEE TO ELECT JOHN RUSSELL TO CONGRESS         2005   
2    C00413674                     STEVE YOUNG FOR CONGRESS         2005   
3    C00193433                                 EMILY'S LIST         2005   
4    C00193433                                 EMILY'S LIST         2005   

  report_type  image_number line_number transaction_id  file_number  \
0          Q3   25971165571          17     SB17.16859       188839   
1          YE   26990167771          17      SB17.4163       198471   
2         12R   25971545941          17         B17142       192557   
3          M9   26980097606         21B    SB21B-77321       201615   
4         M12   26960130301         21B    SB21B-80415       221189   

  entity_type entity_type_desc unused_recipient_committee_id  \
0         ORG     ORGANIZATION                           NaN   
1         IND       INDIVIDUAL                           NaN   
2         ORG     ORGANIZATION                           NaN   
3         ORG     ORGANIZATION                           NaN   
4         ORG     ORGANIZATION                           NaN   

  recipient_committee_id                   recipient_name recipient_state  \
0                    NaN         PROMOTION PARTNERS  INC.              MO   
1                    NaN               MCKINLEY, ILONA MS              FL   
2                    NaN    COMMUNICATIONS BY DESIGN INC.              CA   
3                    NaN  STARFISH DESIGN CAROLYN M. COON              MD   
4                    NaN            UNITED PARCEL SERVICE              PA   

  beneficiary_committee_name national_committee_nonfederal_account  \
0                        NaN                                   NaN   
1                        NaN                                   NaN   
2                        NaN                                   NaN   
3                        NaN                                   NaN   
4                        NaN                                   NaN   

  disbursement_type disbursement_type_description   disbursement_description  \
0               NaN                           NaN     COMPUTER GRAPHICS  ETC   
1               NaN                           NaN  IN-KIND - GRAPHICS DESIGN   
2               NaN                           NaN             GRAPHIC DESIGN   
3               NaN                           NaN            DESIGN/GRAPHICS   
4               NaN                           NaN            DESIGN/GRAPHICS   

  memo_code  memo_code_full    disbursement_date  disbursement_amount  \
0       NaN             NaN  2005-08-15 00:00:00              1300.00   
1       NaN             NaN  2005-12-28 00:00:00               500.00   
2       NaN             NaN  2005-11-16 00:00:00               237.50   
3       NaN             NaN  2005-08-18 00:00:00               168.75   
4       NaN             NaN  2005-11-17 00:00:00               305.35   

  candidate_office candidate_office_description  candidate_office_district  \
0              NaN                          NaN                        NaN   
1                H                        HOUSE                        5.0   
2              NaN                          NaN                        NaN   
3              NaN                          NaN                        NaN   
4              NaN                          NaN                        NaN   

  candidate_id             candidate_name candidate_first_name  \
0          NaN                        NaN                  NaN   
1          NaN  RUSSELL^JOHN TIMOTHY^MR.^    JOHN TIMOTHY MR.    
2          NaN                        NaN                  NaN   
3          NaN                        NaN                  NaN   
4          NaN                        NaN                  NaN   

  candidate_last_name candidate_middle_name candidate_prefix ca

In [21]:
len(df_graphic)

15709

In [26]:
frames = [df_website, df_logo, df_design, df_graphic]

In [30]:
df = pd.concat(frames)

In [35]:
df = df.drop_duplicates()

In [36]:
df.head()

committee_id                              committee_name  report_year  \
0    C00394395                       HUNZEKER FOR CONGRESS         2004   
1    C00300921  BLACK AMERICA'S POLITICAL ACTION COMMITTEE         2004   
2    C00399907                       NEWBERRY FOR CONGRESS         2004   
3    C00365015            COMMITTEE TO ELECT LINCOLN DAVIS         2004   
4    C00392266                      STREUSAND FOR CONGRESS         2004   

  report_type  image_number line_number transaction_id  file_number  \
0          Q2   24962140808          17       EXP:B:71       131870   
1          Q3   25980146785         21B    SB21B.37070       158644   
2         12P   24961973931          17      SB17.4856       129227   
3          Q3   24981789481          17     SB17.13726       156587   
4         12R   24990911003          17      SB17.4601       113365   

  entity_type entity_type_desc unused_recipient_committee_id  \
0         ORG     ORGANIZATION                           NaN   
1         ORG     ORGANIZATION                           NaN   
2         IND       INDIVIDUAL                           NaN   
3         IND       INDIVIDUAL                           NaN   
4         ORG     ORGANIZATION                           NaN   

  recipient_committee_id                    recipient_name recipient_state  \
0                    NaN  DESIGNED INTERNET SOLUTIONS INC.              CA   
1                    NaN                          NGEN LLC              DC   
2                    NaN                NEWBERRY, STIRLING              MA   
3                    NaN                       HAYDEN, TOM              TN   
4                    NaN                     IP PROMOTIONS              TX   

  beneficiary_committee_name national_committee_nonfederal_account  \
0                        NaN                                   NaN   
1                        NaN                                   NaN   
2                        NaN                                   NaN   
3                        NaN                                   NaN   
4                        NaN                                   NaN   

  disbursement_type disbursement_type_description  \
0               NaN                           NaN   
1               NaN                           NaN   
2               NaN                           NaN   
3               NaN                           NaN   
4               NaN                           NaN   

              disbursement_description memo_code  memo_code_full  \
0                              WEBSITE       NaN             NaN   
1         WEBSITE DEVELOPMENT SERVICES       NaN             NaN   
2  WEBSITE DEVELOPMENT AND MAINTENANCE       NaN             NaN   
3                           WEB DESIGN       NaN             NaN   
4                           WEB DESIGN       NaN             NaN   

     disbursement_date  disbursement_amount candidate_office  \
0  2004-05-25 00:00:00               100.00              NaN   
1  2004-09-24 00:00:00                38.33              NaN   
2  2004-07-06 00:00:00              1300.00              NaN   
3  2004-09-01 00:00:00              1000.00                H   
4  2004-03-19 00:00:00              1597.60                H   

  candidate_office_description  candidate_office_district candidate_id  \
0                          NaN                        NaN          NaN   
1                          NaN                        NaN          NaN   
2                          NaN                        NaN          NaN   
3                        HOUSE                        4.0    H4TN06051   
4                        HOUSE                       10.0          NaN   

           candidate_name candidate_first_name candidate_last_name  \
0                     NaN                  NaN                 NaN   
1                     NaN                  NaN                 NaN   
2                     NaN                  NaN                 NaN   
3         D

In [39]:
len(df)

144892

-----

##### Using WashPo's FastFec API

https://github.com/washingtonpost/FastFEC/tree/develop/python

In [15]:
# pip install fastfec

In [16]:
from fastfec import FastFEC

{filing_id}_{form_type}

In [17]:
# from fastfec import FastFEC
# with open('14823832.fec', 'rb') as f:
#     with FastFEC() as fastfec:
#         for form, line in fastfec.parse(f):
#             print("GOT", form, line)

Fast FEC